src/main/kotlin/LoggerUtils.kt

```diff
@@ -0,0 +1,207 @@
+    val throwableProxy = event.throwableProxy
+    return if (throwableProxy != null) {
+        val stackTrace = ThrowableProxyUtil.asString(throwableProxy)
+        val oneLineTrace = inlineStackTrace(stackTrace)
+        "\"exception\":\"$oneLineTrace\", "
+    }
+    else {
+        ""
+    }
+}
+
+fun getDurationInMilliseconds(epochTime: Long): String {
+    val elapsedMilliseconds = epochTime - LogConfiguration.start_time_milliseconds
+    return elapsedMilliseconds.toString()
+}
+
+class JsonLoggerWrapper(private val delegateLogger: Logger) {
+
+    companion object {
+        fun getLogger(forClassName: String): JsonLoggerWrapper {
+            val slf4jLogger: Logger = LoggerFactory.getLogger(forClassName)
+            return JsonLoggerWrapper(slf4jLogger)
+        }
+    }
+
+    fun debug(message: String, vararg tuples: String) {
+        if (delegateLogger.isDebugEnabled) {
+            val semiFormatted = semiFormattedTuples(message, *tuples)
+            delegateLogger.debug(semiFormatted)
+        }
+    }
+
+    fun info(message: String, vararg tuples: String) {
+        if (delegateLogger.isInfoEnabled) {
+            val semiFormatted = semiFormattedTuples(message, *tuples)
+            delegateLogger.info(semiFormatted)
+        }
+    }
+
+    fun warn(message: String, vararg tuples: String) {
+        if (delegateLogger.isWarnEnabled) {
+            val semiFormatted = semiFormattedTuples(message, *tuples)
+            delegateLogger.warn(semiFormatted)
+        }
+    }
+
+    fun error(message: String, vararg tuples: String) {
+        if (delegateLogger.isErrorEnabled) {
+            val semiFormatted = semiFormattedTuples(message, *tuples)
+            delegateLogger.error(semiFormatted)
+        }
+    }
+
+    fun error(message: String, error: Throwable, vararg tuples: String) {
+        if (delegateLogger.isErrorEnabled) {
+            val semiFormatted = semiFormattedTuples(message, *tuples)
+            delegateLogger.error(semiFormatted, error)
+        }
+    }
+}
+
+class LoggerLayoutAppender : LayoutBase<ILoggingEvent>() {
+
+    override fun doLayout(event: ILoggingEvent?): String {
+        if (event == null) {
+            return ""
+        }
+        val dateTime = formattedTimestamp(event.timeStamp)
+        val builder = StringBuilder()
+        builder.append("{ ")
+        builder.append("\"timestamp\":\"")
+        builder.append(dateTime)
+        builder.append("\", \"log_level\":\"")
+        builder.append(event.level)
+        builder.append("\", \"message\":\"")
+        builder.append(flattenMultipleLines(event.formattedMessage))
+        builder.append("\", ")
+        builder.append(throwableProxyEventToString(event))
+        builder.append("\"thread\":\"")
+        builder.append(event.threadName)
+        builder.append("\", \"logger\":\"")
+        builder.append(event.loggerName)
+        builder.append("\", \"duration_in_milliseconds\":\"")
```

Comment:
Can remove duration_in_milliseconds and the method below, it's not a thing in K2HB.

Reply:
Or keep it - I'd like to know what we think the poll time etc is

Reply:
Yeah agreed if it is set and looks good

Reply:
I'll leave it in as at least 1 person thinks it will be useful.